kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -171,36 +173,50 @@ private static void create8021xTls(NetworkProperties props, String deviceId, Map
         String identity = props.get(String.class, "net.interface.%s.config.802-1x.identity", deviceId);
         settings.put("identity", new Variant<>(identity));
 
-        String clientCert = props.get(String.class, "net.interface.%s.config.802-1x.client-cert", deviceId);
-        settings.put("client-cert", new Variant<>(clientCert.getBytes(StandardCharsets.UTF_8)));
+        try {
+            Certificate clientCert = props.get(Certificate.class, "net.interface.%s.config.802-1x.client-cert-name",
+                    deviceId);
+            settings.put("client-cert", new Variant<>(clientCert.getEncoded()));
+        } catch (Exception e) {
+            logger.error("Unable to find or decode Client Certificate", e);
+        }
 
-        String privateKey = props.get(String.class, "net.interface.%s.config.802-1x.private-key", deviceId);
-        settings.put("private-key", new Variant<>(privateKey.getBytes(StandardCharsets.UTF_8)));
+        try {
+            PrivateKey privateKey = props.get(PrivateKey.class, "net.interface.%s.config.802-1x.private-key-name",
+                    deviceId);
+            settings.put("private-key", new Variant<>(privateKey.getEncoded()));
+        } catch (Exception e) {
+            logger.error("Unable to find or decode Private Key", e);
+        }
 
-        String privateKeyPassword = props
-                .get(Password.class, "net.interface.%s.config.802-1x.private-key-password", deviceId).toString();
-        settings.put("private-key-password", new Variant<>(privateKeyPassword));
+        Optional<Password> privateKeyPassword = props.getOpt(Password.class,
+                "net.interface.%s.config.802-1x.private-key-password", deviceId);
+
+        privateKeyPassword.ifPresent(value -> settings.put("private-key-password", new Variant<>(value.toString())));
+
+        settings.put("private-key-password-flags", new Variant<>(new UInt32(4)));
```

Comment:
I would use a constant here for clarity. From the [NM documentation I found](https://manpages.ubuntu.com/manpages/trusty/man5/nm-settings.5.html) that:

![image](https://github.com/eclipse/kura/assets/22748355/b5ee7f1e-beff-4b1b-bda8-2c509dc1462e)

Therefore I would add something like:

```suggestion
        static final UInt32 NM_SECRET_FLAGS_NOT_REQUIRED = new UInt32(4);
        settings.put("private-key-password-flags", new Variant<>(NM_SECRET_FLAGS_NOT_REQUIRED));
```

Reply:
Also: is this correct? I would expect this to be set to `none` as the system should be responsible for providing and storing the secret 🤔 ... this also depends on who is the "user" here 😆